prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -169,4 +180,58 @@ object FHIRBundleHelpers {
             is Observation -> getDiagnosticReportNoObservations(this).forEach { this.deleteResource(it) }
         }
     }
+
+    /**
+     * Evaluates the conditionFilter to get the individual LOINC codes from it
+     * @param fhirBundle the bundle will hold the actual value that %testPerformedCodes will evaluate to
+     * @param receiver the receiver is used to get the condition filter
+     * @param shortHandLookupTable used to get the %testPerformedCodes and its associated fhir path
+     * @return is a list of the LOINC codes as strings
+     */
+    internal fun getConditionsForReceiver(
+        fhirBundle: Bundle,
+        receiver: Receiver,
+        shortHandLookupTable: MutableMap<String, String>
+    ): List<String> {
+        if (receiver.conditionFilter.isNotEmpty()) {
+            val conditions = mutableListOf<String>()
+            FhirPathUtils.pathEngine.evaluate(
+                CustomContext(fhirBundle, fhirBundle, shortHandLookupTable),
+                fhirBundle,
+                fhirBundle,
+                fhirBundle,
+                FhirPathUtils.parsePath(receiver.conditionFilter[0])
+            ).forEach {
+                conditions.add(it.toString())
+            }
+            return conditions
+        }
+        return listOf()
+    }
+
+    /**
+     * Adds the observation for those that have a condition that matches the condition filter
+     * @param fhirBundle the bundle to look for matching observations in
+     * @param conditions the conditions to match on
+     * @return is a list of the extensions which will be added to the bundle
+     */
+    internal fun addObservationsToReceivers(fhirBundle: Bundle, conditions: List<String>): List<Extension> {
```

Comment:
You need to do this differently to be able to support any FHIR Path expression. You should be able to look through all the observations, set them as the focus resource, then evaluate the condition filters to see if that observation needs to be added.